mephisto/abstractions/providers/prolific/prolific_unit.py

```diff
@@ -0,0 +1,365 @@
+            # These statuses only change on agent dependent changes
+            agent = self.get_assigned_agent()
+            found_status = self.db_status
+
+            if agent is not None:
+                agent_status = agent.get_status()
+                if agent_status == AgentState.STATUS_APPROVED:
+                    found_status = AssignmentState.ACCEPTED
+                elif agent_status == AgentState.STATUS_REJECTED:
+                    found_status = AssignmentState.REJECTED
+                elif agent_status == AgentState.STATUS_SOFT_REJECTED:
+                    found_status = AssignmentState.SOFT_REJECTED
+            else:
+                logger.warning(f'Agent for unit {self} is None')
+
+            if found_status != self.db_status:
+                self.set_db_status(found_status)
+
+            return self.db_status
+
+        # Remaining statuses are tracking a live Study
+        prolific_study_id = self.get_prolific_study_id()
+        if prolific_study_id is None:
+            # If the study_id is None and there's an agent still assigned,
+            # then that agent has timed out, and we should expire
+            agent = self.get_assigned_agent()
+            if agent is not None:
+                if agent.get_status() != AgentState.STATUS_EXPIRED:
+                    agent.update_status(AgentState.STATUS_EXPIRED)
+
+            # Can't determine anything else if there is no Study on this unit
+            return self.db_status
+
+        requester: 'ProlificRequester' = self.get_requester()
+        client = self._get_client(requester.requester_name)
+        study = prolific_utils.get_study(client, prolific_study_id)
+
+        if study is None:
+            return AssignmentState.EXPIRED
+
+        local_status = self.db_status
+        external_status = self.db_status
+
+        if study.status == StudyStatus.UNPUBLISHED:
+            external_status = AssignmentState.COMPLETED
+        elif study.status == StudyStatus.ACTIVE:
+            external_status = AssignmentState.LAUNCHED
+        elif study.status == StudyStatus.SCHEDULED:
+            # TODO (#1008): Choose correct mapping
+            pass
+        elif study.status == StudyStatus.PAUSED:
+            # TODO (#1008): Choose correct mapping
+            pass
+        elif study.status == StudyStatus.AWAITING_REVIEW:
+            # TODO (#1008): Choose correct mapping
+            external_status = AssignmentState.COMPLETED
+            pass
+        elif study.status == StudyStatus.COMPLETED:
+            external_status = AssignmentState.COMPLETED
+        else:
+            raise Exception(f'Unexpected Study status {study.status}')
+
+        # -------------------------------------------------------------------------
+        # TODO (#1008): Fix this logic. It looked abstract and unclear in Mturk code.
+        if external_status != local_status:
+            if local_status == AssignmentState.ASSIGNED and external_status in [
```

Comment:
if locally we think it's assigned, but prolific thinks it's launched, it must have been returned (or expired).

Reply:
Ah no, I'm explaining the logic. You're saying to fix it in the TODO